TreeMaker/interface/TreeMaker.h

```diff
@@ -356,3 +358,110 @@ class TreeRecoCand : public TreeObject<vector<TLorentzVector> > {
 		bool doLorentz{};
 		vector<double> pt, eta, phi, energy;
 };
+
+// Derived version of vector<vector<T>> with switch for vector<T> values and vector<int> offsets instead
+template <typename Base = double, typename Sub = std::vector<Base>, typename Top = std::vector<Sub>> 
```

Comment:
in what case would we want Sub *not* to be `std::vector<Base>`?

I think Sub and Top should not be template parameters, but typedefs inside the class:
```c++
typedef Sub std::vector<Base>;
typedef Top std::vector<Sub>;
```

Reply:
this means you'll have to specify ` : public TreeObject<std::vector<std::vector<Base>>>` in the class definition, but oh well

Reply:
Done